x-pack/plugins/security/server/lib/authorization/check_privileges.js

```diff
@@ -59,13 +61,15 @@ export function checkPrivilegesWithRequestFactory(shieldClient, config, actions,
           }],
           index: [{
             names: [kibanaIndex],
-            privileges: ['create', 'delete', 'read', 'view_index_metadata']
+            privileges: buildLegacyIndexPrivileges()
           }],
         }
       });
 
+      validateEsPrivilegeResponse(hasPrivilegesResponse, application, allApplicationPrivileges, [ALL_RESOURCE], kibanaIndex);
```

Comment:
The tests aren't ensuring that we're calling this...

If we use ` jest.mock('')` to stub out the static import, we can ensure that this is being called with the proper arguments; or we can mock out the ES response with some complete bogus responses and make sure that at least those are rejecting with the validation errors, and leave the more fine-grained permutation based testing for the `validateEsPrivilegeResponse` tests. I haven't really found a way that I prefer in all situations yet...